tmva/sofie/inc/TMVA/ROperator_Shape.hxx

```diff
@@ -0,0 +1,83 @@
+#ifndef TMVA_SOFIE_ROPERATOR_Shape
+#define TMVA_SOFIE_ROPERATOR_Shape
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <sstream>
+#include <iostream>
+#include<sstream>
+#include<vector>
+#include <iterator>
+#include<string>
+namespace TMVA{
+namespace Experimental{
+namespace SOFIE{
+
+template <typename T>
+class ROperator_Shape final : public ROperator
+{
+
+private:
+
+   /* Attributes*/
+   std::string fNX;
+   std::string fNY;
+   std::string result;
+   std::vector<size_t> fShape;
+   std::vector<size_t> output_shape;
+   std::string fType;
+
+public:
+   ROperator_Shape(){}
+   ROperator_Shape(std::string nameX, std::string nameY):
+   fNX(UTILITY::Clean_name(nameX)), fNY(UTILITY::Clean_name(nameY)){}
+
+   std::vector<ETensorType> TypeInference(std::vector<ETensorType> input){
+      return input;
+   }
+
+   std::vector<std::vector<size_t>> ShapeInference(std::vector<std::vector<size_t>> input){
+      std::vector<std::vector<size_t>>  ret;
+      ret[0].push_back(input.size()); 
+      ret[0].push_back(input[0].size());
+      return ret;
+   }
+
+   void Initialize(RModel& model){
+      if (model.CheckIfTensorAlreadyExist(fNX) == false){   //input must be a graph input, or already initialized intermediate tensor
+         throw std::runtime_error("TMVA SOFIE Shape Op Input Tensor is not found in model");
+      }
+      fShape = model.GetTensorShape(fNX);
+      result = ConvertShapeToString(fShape);
```

Comment:
This code here is not correct and it causes the problem. 
The output shape is dim=1 and is equal to input_length.
For example if input is a tensor with shape {1,2} output is a tensor with shape  {2} containing 1 and 2 as values.